eng/testing/tests.mobile.targets

```diff
@@ -146,6 +157,32 @@
 
   </Target>
 
+  <Target Condition="'$(EnableAggressiveTrimming)' == 'true'"
+          Name="ConfigureTrimming"
+          BeforeTargets="PrepareForILLink">
+    <PropertyGroup>
+      <TrimMode>link</TrimMode>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <ManagedAssemblyToLink Include="$(OutDir)\*xunit*">
+        <IsTrimmable>true</IsTrimmable>
+        <TrimMode>link</TrimMode>
+      </ManagedAssemblyToLink>
+      <ManagedAssemblyToLink Condition="('%(ManagedAssemblyToLink.FileName).dll' != '$(MSBuildProjectName).dll')" >
+        <TrimMode>link</TrimMode>
+      </ManagedAssemblyToLink>
+      <ManagedAssemblyToLink Condition="('%(ManagedAssemblyToLink.FileName).dll' == '$(AssemblyName).dll')" >
+        <TrimMode>copy</TrimMode>
+      </ManagedAssemblyToLink>
```

Comment:
Looking at https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L142-L144 , I think instead of this, we should have a target that adds to `@(ResolvedFileToPublish)`, and sets the `TrimMode` metadata as needed.

This will get processed by https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L235-L237 and subsequently by [PrepareForILLink](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L146) target.

- IIUC, you just need to add the xunit files, if they aren't there already, and set `TrimMode=link` for them.
- And make sure that the test assembly doesn't have `IsTrimmable`, or `TrimMode` set. [this](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L211-L214) should handle the rest for you.

Reply:
Using ResolvedFileToPublish makes sense if we can hook it early enough. Though I'd keep copy setting to be explicit and maybe add a comment that we want the tested assembly to be always copied only.

Reply:
Will address in a follow-up PR